src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java

```diff
@@ -2,31 +2,83 @@
 import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
 
 import javax.annotation.Resource;
+import java.util.List;
 
+// TODO: Move these services to StudyResource
 @Api(value = "Study Germplasm Services")
 @Controller
 @RequestMapping("/crops")
 public class StudyGermplasmResource {
 
-    @Resource
-    private StudyGermplasmService studyGermplasmService;
+	@Resource
+	private StudyGermplasmService studyGermplasmService;
+
+	@ApiOperation(value = "Replace germplasm entry in study",
+		notes = "Replace germplasm entry in study")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/entries/{entryId}", method = RequestMethod.PUT)
+	@ResponseBody
+	public ResponseEntity<StudyGermplasmDto> replaceStudyGermplasm(final @PathVariable String cropname,
+		@PathVariable final String programUUID,
+		@PathVariable final Integer studyId, @PathVariable final Integer entryId, @RequestBody final StudyGermplasmDto studyGermplasmDto) {
+		return new ResponseEntity<>(this.studyGermplasmService.replaceStudyGermplasm(studyId, entryId, studyGermplasmDto),
+			HttpStatus.OK);
+
+	}
+
+	@ApiOperation(value = "Create germplasm entries in study based on the specified germplasm list",
+		notes = "Create germplasm entries in study based on the specified germplasm list")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/entries/generation", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<List<StudyGermplasmDto>> createStudyGermplasm(final @PathVariable String cropname,
+		@PathVariable final String programUUID,
+		@PathVariable final Integer studyId, @RequestBody final GermplasmEntryRequestDto germplasmEntryRequestDto) {
+
+		return new ResponseEntity<>(
+			this.studyGermplasmService.createStudyGermplasmList(studyId, germplasmEntryRequestDto.getGermplasmListId()),
+			HttpStatus.OK);
+
+	}
+
+	@ApiOperation(value = "Delete germplasm entries in study",
+		notes = "Delete germplasm entries in study")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/entries", method = RequestMethod.DELETE)
+	@ResponseBody
+	public ResponseEntity deleteStudyGermplasm(final @PathVariable String cropname,
+		@PathVariable final String programUUID,
+		@PathVariable final Integer studyId) {
+
+		this.studyGermplasmService.deleteStudyGermplasm(studyId);
+
+		return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+
+	}
+
+	@ApiOperation(value = "Update germplasm entry property",
+		notes = "Update germplasm entry property")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/entries", method = RequestMethod.PUT)
```

Comment:
@abatac this URL refers to the entry entity when it is updating a property. 
/{cropname}/programs/{programUUID}/studies/{studyId}/entries/{entryId}/properties/{propertyId}
should work better